### PR TITLE
Add estimator to provider

### PR DESCRIPTION
**Pull request by jyu00**
### Summary

This PR adds a new method `estimate` to the provider that's similar to `run_circuits`. It calls the corresponding runtime program under the cover.

### CODE REVIEW

File: qiskit/providers/ibmq/accountprovider.py [hunk 1]
```diff
@@ -320,6 +323,68 @@ def run_circuits(
...
 
+    def estimate(
+            self,
+            state: Union[QuantumCircuit, Statevector],
+            observable: Union[BaseOperator, PauliSumOp, List[Tuple[str, float]]],
+            parameters: Union[List[Union[float, List[float]]]] = None,
+            evaluator: Optional[str] = None,
+            transpile_options: Optional[dict] = None,
+            shots: Optional[int] = None,
+            backend: Optional[Union[Backend, BaseBackend, str]] = None,
```
Comment (dlasecki) [inline qiskit/providers/ibmq/accountprovider.py:334]: Should not be optional because the value of `None` raises an error below.

**Comment by nonhermitian** [comment]
This is not a great name for an expectation value routine.

File: qiskit/providers/ibmq/accountprovider.py [hunk 2]
```diff
@@ -320,6 +323,68 @@ def run_circuits(
...
+        """Estimate the expectation value.
+
+        Args:
+            state: Quantum circuit that represents a quantum state.
+
+            observable: The Hamiltonian to be evaluated.
+
+            parameters: The parameters to be bound.
+
+            evaluator: The name of the evaluator class.
```
Comment (ajavadia) [inline qiskit/providers/ibmq/accountprovider.py:346]:
```suggestion
            evaluator: The name of the evaluator class. Defaults to 'PauliExpectationValue'.
```

File: qiskit/providers/ibmq/accountprovider.py [hunk 3]
```diff
@@ -320,6 +323,68 @@ def run_circuits(
...
+
+            backend: Backend to execute circuits on.
+                Transpiler options are automatically grabbed from backend configuration
+                and properties unless otherwise specified.
+
+            transpile_options: Additional transpiler options.
```
Comment (ajavadia) [inline qiskit/providers/ibmq/accountprovider.py:352]:
```suggestion
            transpile_options: Additional transpiler options. If not specified, default Qiskit transpilation to the backend is used.
```

File: qiskit/providers/ibmq/accountprovider.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jyu00) [reply qiskit/providers/ibmq/accountprovider.py]: Done in 3a3d88d.

**Pull request closed without merging**